Makefile

```diff
@@ -1,5 +1,52 @@
+all: test check-coverage lint tidy ## Tests, lints and checks coverage
+
+clean:  ## Remove generated files
+
+.PHONY: all clean
+
+# -- Lint ----------------------------------------------------------------------
+lint: ## Lint Go Source Code
+	golangci-lint run
+
+tidy:
+	go mod tidy
+
+.PHONY: lint tidy clean
+
+# -- Test ----------------------------------------------------------------------
 COVERFILE=coverage.out
+COVERAGE = 50
+
+test:
+	go test -coverprofile=$(COVERFILE) -tags codeanalysis ./...
+
+check-coverage: test  ## Check that test coverage meets the required level
+	@go tool cover -func=$(COVERFILE) | $(CHECK_COVERAGE) || $(FAIL_COVERAGE)
+
+coverage: test  ## Show test coverage in your browser
+	go tool cover -html=$(COVERFILE)
+
+clean: 
+	rm -f $(COVERFILE)
+
+CHECK_COVERAGE = awk -F '[ \t%]+' '/^total:/ && $$3 < $(COVERAGE) {exit 1}'
+FAIL_COVERAGE = { echo '$(COLOUR_RED)FAIL - Coverage below $(COVERAGE)%$(COLOUR_NORMAL)'; exit 1; }
 
+.PHONY: check-coverage coverage test
+
+# --- Utilities ---------------------------------------------------------------
+COLOUR_NORMAL = $(shell tput sgr0 2>/dev/null)
+COLOUR_RED    = $(shell tput setaf 1 2>/dev/null)
+COLOUR_GREEN  = $(shell tput setaf 2 2>/dev/null)
+COLOUR_WHITE  = $(shell tput setaf 7 2>/dev/null)
+BOLD          = $(shell tput bold 2>/dev/null)
+
+help:
+	@awk -F ':.*## ' 'NF == 2 && $$1 ~ /^[A-Za-z0-9_-]+$$/ { printf "$(BOLD)$(COLOUR_WHITE)%-20s$(COLOUR_NORMAL)%s\n", $$1, $$2}' $(MAKEFILE_LIST) | sort
```

Comment:
I think it's great that you are re-using this `help` target.

Maybe also add help messages for all targets with `##`.
You can look at the output with `make help`

Reply:
Thanks for that, added help messages for all targets, it's a really nice feature to have in a makefile!